src/graph/service/PasswordAuthenticator.cpp

```diff
@@ -13,7 +13,16 @@ PasswordAuthenticator::PasswordAuthenticator(meta::MetaClient* client) {
 }
 
 Status PasswordAuthenticator::auth(const std::string& user, const std::string& password) {
```

Comment:
Don't modify `auth` function, do it outside.

Reply:
I need to pass the client addr into `MetaClient::authCheckFromCache()`, and `auth` is the middleware

Reply:
You could add `clientIp` to meta client in `GraphService::auth`

Reply:
I have passed clientIp in [src/graph/service/GraphService.cpp line 122](https://github.com/vesoft-inc/nebula/pull/3942/files#diff-3a9c8c30a7bc9f5d28cd53b6f7f2f3b29a67827996e45f6e6bd2d2a749db19c5R212)
Besides, the  `GraphService::auth()` in the enterprise version also has `clientIp` as parameter. This should be fine